bigflow/build/operate.py

```diff
@@ -72,14 +79,57 @@ def _export_docker_image_to_file(tag: str, target_dir: Path, version: str):
     bf_commons.run_process(["docker", "image", "save", "-o", image_target_path, bf_commons.get_docker_image_id(tag)])
 
 
-def _build_docker_image(project_dir: Path, tag: str):
+def _build_docker_image(
+    project_spec: BigflowProjectSpec,
+    tag: str,
+    cache_params: BuildImageCacheParams | None,
+):
+
     logger.debug("Run docker build...")
-    bf_commons.run_process(f'docker build {project_dir} --tag {tag}')
+    cmd = ["docker", "build", project_spec.project_dir, "--tag", tag]
+
+    if cache_params:
+
+        logger.debug("Authenticate to docker registry")
+        bigflow.deploy.authenticate_to_registry(
+            auth_method=cache_params.auth_method or bigflow.deploy.AuthorizationType.LOCAL_ACCOUNT,
+            vault_endpoint=cache_params.vault_endpoint,
+            vault_secret=cache_params.vault_secret,
+        )
+
+        image = cache_params.cache_from_image or ""
+        if cache_params.cache_from_version:
+            image += f",{project_spec.docker_repository}:{cache_params.cache_from_version}"
```

Comment:
Can we move those two lines to some method with a significant name?

Reply:
IMO it's not very beneficial to move those 4 lines into a separate function (adds 4 more lines + extra view jump for a reader). But I'm not against this change.

Moved this into a new function.

Reply:
👍